### PR TITLE
Custom, more flexible, check for declaration order.

### PR DESCRIPTION
**Pull request by Chris-V**

### CODE REVIEW

File: checkstyle-base/src/main/java/com/arcbees/checkstyle/checks/Defaults.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import static com.arcbees.checkstyle.checks.modifiers.Visibility.PACKAGE_PRIVATE;
+import static com.arcbees.checkstyle.checks.modifiers.Visibility.PRIVATE;
+import static com.arcbees.checkstyle.checks.modifiers.Visibility.PROTECTED;
+import static com.arcbees.checkstyle.checks.modifiers.Visibility.PUBLIC;
+import static com.arcbees.checkstyle.checks.representation.Declaration.newDeclaration;
+
+public final class Defaults {
+    private static final boolean ONE_ERROR_PER_FILE = false;
+
+    private static final List<Declaration> DECLARATION_ORDER = Arrays.asList(
```
Comment (Chris-V) [inline checkstyle-base/src/main/java/com/arcbees/checkstyle/checks/Defaults.java]: Please review the default order.

**Comment by christiangoudreau** [comment]
LGTM

**Comment by Pacane** [comment]
LGTM

File: checkstyle-base/src/main/java/com/arcbees/checkstyle/checks/Defaults.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            newDeclaration(CLASS).withStatic(STATIC).build(),
+            newDeclaration(CLASS).withStatic(NOT_STATIC).build(),
+
+            newDeclaration(FIELD).withStatic(STATIC).withFinal(FINAL).build(),
+            newDeclaration(FIELD).withStatic(STATIC).withFinal(NOT_FINAL).build(),
+
+            newDeclaration(INITIALIZER).withStatic(STATIC).build(),
+
+            newDeclaration(FIELD).withStatic(NOT_STATIC).withFinal(FINAL).build(),
+            newDeclaration(FIELD).withStatic(NOT_STATIC).withFinal(NOT_FINAL).build(),
```
Comment (meriouma) [inline checkstyle-base/src/main/java/com/arcbees/checkstyle/checks/Defaults.java]: Maybe we should consider visibility for fields? I understand the argument for methods but I would be better to have fields sorted by visibility

**Comment by meriouma** [comment]
LGTM

File: checkstyle-base/src/main/java/com/arcbees/checkstyle/checks/Defaults.java (hunk 2 above)
Comment (christiangoudreau) [inline checkstyle-base/src/main/java/com/arcbees/checkstyle/checks/Defaults.java]: It should and it must!

Comment (Chris-V) [inline checkstyle-base/src/main/java/com/arcbees/checkstyle/checks/Defaults.java]: Updated. Still good?

**Comment by christiangoudreau** [comment]
LGTM

**Pull request merged**